### DISCUSSION TITLE
Message time attribute

### DISCUSSION THREAD

**Discussion by ghost** [mido/mido#417] [Q&A] [answered]
What does actually the time attribute of a Message? Setting it to different values doesn't change anything. And more specifically how can i send a note_off message after a certain delay?

**Comment by olemb** [comment, marked as answer]
The `time` attribute is something I added just for convenience and is mostly not used by Mido for anything. I have a summary here:

    https://mido.readthedocs.io/en/latest/messages.html#the-time-attribute

Sorry for the confusion.

As for sending a message after a specified time there is nothing built into Mido for doing this. `port.send()` will ignore `time` and send the message right away. I am trying to think of a good way to do this, but I think the best way would be to use a system with an event loop such as Tkinter or something built on asyncio. In these you can say things like `after(specified_time, port.send, msg)` or `at(exact_time, dome_something)`.

If you like I can have a look for some old code where I did this. (I remember creating a program that echoed what you play after 0.5 seconds on another synthesizer, but it was in 2006 so I can't remember the library I used.)
